### PR TITLE
configure deployment to any portal.

### PR DESCRIPTION
**Pull request by dknochen**

### CODE REVIEW

File: tasks/config.js [hunk 1]
```diff
@@ -129,10 +131,18 @@ gulp.task('load-config', ['config-template'], function(done){
         config.bucketName = config.isOnProduction ? config.credentials.s3.bucketPrefix : config.credentials.s3.bucketPrefix + '-' + config.buildId;
         config.projectName = config.isOnProduction ? config.credentials['28'].projectPrefix : config.credentials['28'].projectPrefix + '-' + buildId;
 
+        // where to deploy the cellstore?
+        config.portalAPIUrl = config.credentials['28'].protocol ? config.credentials['28'].protocol : 'http';
```
Comment (wcandillon) [inline tasks/config.js]: @dknochen `_.defaults` might be cleaner here.

File: tasks/config.js [hunk 2]
```diff
@@ -129,10 +131,18 @@ gulp.task('load-config', ['config-template'], function(done){
...
+        config.portalAPIUrl += '://' + (config.credentials['28'].portalProject ? config.credentials['28'].portalProject : 'portal');
```
Comment (wcandillon) [inline tasks/config.js]: @dknochen use `_.template`

**Comment by wcandillon** [comment]
@dknochen `edinet.json.enc` seems unchanged.

**Comment by wcandillon** [comment]
@dknochen `edinet.json` seems unchanged

**Comment by wcandillon** [comment]
@dknochen Beautiful. I have left some comments :information_desk_person:.

**Comment by dknochen** [comment]
@wcandillon the update in edinet.json is not technically needed as it uses the defaults anyway. But its more easy to update this way.

**Comment by wcandillon** [comment]
@dknochen I agree make sense

File: tasks/config.js [hunk 3]
```diff
@@ -129,10 +131,16 @@ gulp.task('load-config', ['config-template'], function(done){
         config.bucketName = config.isOnProduction ? config.credentials.s3.bucketPrefix : config.credentials.s3.bucketPrefix + '-' + config.buildId;
```
Comment (wcandillon) [inline tasks/config.js]: _.default doesn't work here right?

File: tasks/templates/config.json.mustache [hunk 4]
```diff
@@ -40,6 +40,10 @@
         "projectPrefix": "{{{credentials.28-projectPrefix}}}",
         "email": "{{{credentials.28-email}}}",
         "password": "{{{credentials.28-password}}}",
+        "protocol": "{{{credentials.28-protocol}}}{{^credentials.28-protocol}}http{{/credentials.28-protocol}}",
```
Comment (wcandillon) [inline tasks/templates/config.json.mustache]: @dknochen the key name seems unspecific.

File: tasks/config.js [hunk 5]
```diff
@@ -129,10 +131,16 @@ gulp.task('load-config', ['config-template'], function(done){
...
         config.projectName = config.isOnProduction ? config.credentials['28'].projectPrefix : config.credentials['28'].projectPrefix + '-' + buildId;
 
+        // where to deploy the cellstore?
+        config.portalAPIUrl =
+            _.template('<%= protocol %>://<%= portalProject %>.<%= portalDomain %><%= portalApiPrefix %>')(config.credentials['28']);
```
Comment (wcandillon) [inline tasks/config.js]: @dknochen :+1:

**Pull request merged**